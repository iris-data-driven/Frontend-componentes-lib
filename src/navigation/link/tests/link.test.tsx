import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { render, screen, waitFor, fireEvent } from '@testing-library/react';

import TestingLinkApp from './utils/TestingLinkApp';

describe('Link', () => {
  it('Should match the snapshot', () => {
    const text = 'Link text';

    render(
      <Router>
        <TestingLinkApp text={text} />
      </Router>
    );

    const component = screen.getByRole('link');

    expect(component).toMatchSnapshot();
  });

  it('Should render foo text correctly', async () => {
    const text = 'Link text';

    const { container, getByText } = render(
      <Router>
        <TestingLinkApp text={text} />
      </Router>
    );

    expect(container.innerHTML).toMatch('You are home');

    expect(getByText(text)).toBeInTheDocument();

    fireEvent.click(getByText(text));

    await waitFor(() => {
      expect(container.innerHTML).toMatch('You are on the about page');
    });
  });
});
