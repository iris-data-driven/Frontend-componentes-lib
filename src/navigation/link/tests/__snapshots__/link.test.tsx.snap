// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Should match the snapshot 1`] = `
<a
  href="/about"
  role="link"
>
  <p
    class="sc-AxjAm cllMKu"
    role="paragraph"
    type="p2"
  >
    Link text
  </p>
</a>
`;
